sammba/modality_processors/perfusion_fair.py

```diff
@@ -0,0 +1,387 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Sunday 4th March 2018
+
+@author: Nachiket Nadkarni
+"""
+
+
+import pandas as pd
+import nibabel as nib
+import numpy as np
+from scipy.optimize import least_squares as ls
+from multiprocessing import cpu_count, Pool
+from functools import partial
+import tqdm
+
+
+def perf_fair_read_ptbl(nii_fname):
+    """
+    Extract perfusion FAIR acquisition parameters (TIs, order of selective and
+    non-selective inversions) from the *_ptbl.txt file of a *.nii.gz file. Note
+    that both are created simultaneously when sammba.io_conversions.dcm_to_nii
+    is used for DICOM to NIfTI-1 conversion. If a different converter was used
+    that does not export this data, or does it differently, this function will
+    not work.
+
+    Parameters
+    ----------
+    nii_fname : str
+        Path to the perfusion *.nii.gz file converted by
+        sammba.io_conversions.dcm_to_nii, NOT to the *_ptbl.txt file itself.
+        The *_ptbl.txt file name is determined automatically from that of the
+        *.nii.gz.
+
+    Returns
+    -------
+    Dictionary of parameters needed for perfusion fitting:
+    ti:
+        simple list of all TIs
+    long_ti:
+        full list of TIs, the same length and in the same order as the
+        acquisitions (the final dimension of the NIfTI-1 file)
+    fc:
+        full list of inversion type descriptions ("Selective" or
+        "Non-selective"), the same length and in the same order as the
+        acquisitions (the final dimension of the NIfTI-1 file). fc is a DICOM
+        acronym for frame comment, the DICOM field used for storing the

```

Comment:
expanding out acronyms makes code difficult to read.

Reply:
I don't think so. I don't want to spend time guessing what the coder wants to output at each line. Moreover I personally find it hard to follow a code when I need to remember several acronyms because of the memory load.